### DISCUSSION TITLE
@nivo/chord onRibbonClick

### DISCUSSION THREAD

**Discussion by villdar** [plouc/nivo#1698] [Q&A]
Hi! I'm trying to making a Chord chart interactive. I want to click on a ribbon then show some other more specific graphic.

I've tryed build a function on a click but probably I'm missing something.

Any tips or example to do something like i said?

Thanks for any helps!

**Comment by wyze** [comment]
Should be passing a function to `onRibbonClick`, like we do on the website https://github.com/plouc/nivo/blob/master/website/src/pages/chord/index.js#L133. Can you come up with a CodeSandbox of what you've tried and we can take a look?

**Reply by villdar** [reply]
Thanks for the quick answer. I'll take a look at what you sent me. 
Anyway, here's what I'm trying to do. https://codesandbox.io/embed/wandering-smoke-lle4u?fontsize=14&hidenavigation=1&theme=dark
If it hurts I know, I'm learning React and I wanted to play a bit with these graphics.
My idea was to make the ribbons or arcs clickable, and change the graph based on where you click.

**Reply by wyze** [reply]
Here you go: https://codesandbox.io/s/nervous-turing-z7mld

Functional components don't have a `this` context.

**Reply by villdar** [reply]
Thanks you so much :) that's what I wanted!
